test/realm_set_test.dart

```diff
@@ -0,0 +1,535 @@
+      case _NullableDateTime:
+        return Sets(nullableDateTimeSet as RealmSet<DateTime?>, [...setByType(DateTime).values, null]);
+      case _NullableObjectId:
+        return Sets(nullableObjectIdSet as RealmSet<ObjectId?>, [...setByType(ObjectId).values, null]);
+      case _NullableUuid:
+        return Sets(nullableUuidSet as RealmSet<Uuid?>, [...setByType(Uuid).values, null]);
+      default:
+        throw RealmError("Unsupported type $type");
+    }
+  }
+
+  List<Object?> values(Type type) {
+    return setByType(type).values;
+  }
+
+  List<Object?> getValuesOrManagedValues(Realm realm, Type type) {
+    Sets set = setByType(type);
+    if (set.values.first is! RealmObject) {
+      return set.values;
+    }
+
+    return set.values.map<Object?>((value) {
+      RealmObject? realmValue = set.getRealmObject!(realm, value as RealmObject);
+      return realmValue;
+    }).toList();
+  }
+}
+
+class Sets {
+  final RealmSet<Object?> set;
+  final List<Object?> values;
+  RealmObject? Function(Realm realm, RealmObject value)? getRealmObject = (realm, value) => value;
+
+  Sets(this.set, this.values, [this.getRealmObject]);
+}
+
+Future<void> main([List<String>? args]) async {
+  await setupTests(args);
+
+  for (var type in supportedTypes) {
+    test('RealmSet<$type> unmanged set add', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.contains(values.first), true);
+
+      set.remove(values.first);
+      expect(set.length, equals(0));
+      expect(set.contains(values.first), false);
+    });
+
+    test('RealmSet<$type> unmanged set remove', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.contains(values.first), true);
+
+      set.remove(values.first);
+      expect(set.length, equals(0));
+      expect(set.contains(values.first), false);
+    });
+
+    test('RealmSet<$type> unmanged set elementAt', () {
+      final testSet = TestRealmSets(1);
+      final set = testSet.setByType(type).set;
+      final values = testSet.values(type);
+
+      set.add(values.first);
+      expect(set.length, equals(1));
+      expect(set.elementAt(0), values.first);
+    });
+
+    test('RealmSet<$type> create', () {
+      var config = Configuration.local([TestRealmSets.schema, Car.schema]);
+      var realm = getRealm(config);
+
+      var testSet = TestRealmSets(1);
+
+      realm.write(() {
+        realm.add(testSet);
+      });
+
+      expect(realm.find<TestRealmSets>(1), isNotNull);
+
+      testSet = realm.find<TestRealmSets>(1)!;
+      var set = testSet.setByType(type).set;
+
+      expect(set.length, equals(0));
+    });
+
+    test('RealmSet<$type> create from unmanaged', () {
```

Comment:
```suggestion
    test('RealmSet<$type> create set with values', () {
```

Reply:
I will leave as it is. it is creating a managed RealmSet from unmanaged RealmSet